aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -726,19 +726,23 @@ Tensor index_select_mps(const Tensor & self,
           newCachedGraph = new CachedGraph(mpsGraph);
 
           MPSGraphTensor* inputTensor = mpsGraphRankedPlaceHolder(mpsGraph, self);
-          MPSGraphTensor* maskTensor = mpsGraphRankedPlaceHolder(mpsGraph, mask);
+          MPSGraphTensor* maskTensor = mpsGraphRankedPlaceHolder(mpsGraph, *b_mask);
           MPSDataType valueType = getMPSScalarType(value.type());
+          MPSDataType inputDataType = getMPSScalarType(self.scalar_type());
 
           // constantWithScalar doesn't like Bool constants getting created so
-          // mapping them to int8
-          if (valueType == MPSDataTypeBool) {
+          // mapping them to int8.
+          // Starting with macOS 13.0, this cast is not needed anymore
+          if (valueType == MPSDataTypeBool && !MPSDevice::getInstance()->macOS_13_0_or_newer()) {
             valueType = MPSDataTypeInt8;
           }
-          MPSGraphTensor* valueTensor =  [mpsGraph constantWithScalar:value.to<double>()
-                                                            dataType:valueType];
+
+          MPSGraphTensor* valueTensor =
```

Comment:
Can we make a placeholder for value instead of putting it in the key to avoid recreating graphs when `value` changes?

Reply:
yes, good idea